File: js/reverseString.js
const reverseString = (s) => {
	let temp 
	const length = s.length;
	for (i = 0; i < length / 2; i++) {
		temp = s[i]
		s[i] = s[length-1-i]
		s[length-1-i] = temp
	}
}

function main() {
	const input = ["H", "e", "l", "l", "o"]
	reverseString(input)
	console.log(input)
}

main()

File: go/reverse_string.go
package main

import (
	"fmt"
)

func reverseString(s []string) {
	i, j := 0, len(s)-1
	for i < j {
		s[i], s[j] = s[j], s[i]
		i += 1
		j -= 1
	}
}

func main() {
	input := []string{"h", "e", "l", "l", "o"}
	reverseString(input)
	fmt.Println(input)
}
